hack/role/role.go

```diff
@@ -0,0 +1,413 @@
+package main
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"flag"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"os/exec"
+	"path"
+	"regexp"
+	"slices"
+	"strings"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azidentity"
+	sdkauthorization "github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/authorization/armauthorization/v3"
+	cloudcredentialv1 "github.com/openshift/cloud-credential-operator/pkg/apis/cloudcredential/v1"
+	"github.com/sirupsen/logrus"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/yaml"
+
+	"github.com/Azure/ARO-RP/pkg/env"
+	"github.com/Azure/ARO-RP/pkg/util/azureclient/azuresdk/armauthorization"
+	"github.com/Azure/ARO-RP/pkg/util/version"
+)
+
+type permissionSet struct {
+	name             string
+	manifest         string
+	roleDefinitionId string
+}
+
+type manager struct {
+	roleDefinitions armauthorization.RoleDefinitionsClient
+}
+
+type response struct {
+	Tags          []tag `json:"tags"`
+	Page          int   `json:"page"`
+	HasAdditional bool  `json:"has_additional"`
+}
+
+type tag struct {
+	Name         string   `json:"name"`
+	LastModified datetime `json:"last_modified"`
+}
+
+type datetime struct {
+	time.Time
+}
+
+func (d *datetime) UnmarshalJSON(data []byte) error {
+	dd, err := time.Parse(fmt.Sprintf("\"%s\"", time.RFC1123Z), string(data))
+	if err != nil {
+		return err
+	}
+	d.Time = dd
+	return nil
+}
+
+type missingPermissionsError struct{}
+
+func (m *missingPermissionsError) Error() string {
+	return "missing permissions"
+}
+
+type emptyVersionsError struct{}
+
+func (e *emptyVersionsError) Error() string {
+	return "no versions found"
+}
+
+var rxImage = regexp.MustCompile(`\d+\.\d+\.\d+(-(rc|ec).\d+)?-x86_64`)
+
+var (
+	verifiedVersion = flag.String("verified-version", "", "verified version")
+	targetVersion   = flag.String("target-version", "", "(optional) version to be verified")
+	ocBinary        = flag.String("oc-bin", "oc", "path to oc binary")
+	permissionSets  = []permissionSet{
+		{
+			"CloudControllerManager",
```

Comment:
One very small nit, and then we can merge. 

https://github.com/Azure/ARO-RP/pull/3916/files recently changed the operator/key names in RP. Can we make the same changes here?

Reply:
done